src/parquet/file/reader-internal.cc

```diff
@@ -171,16 +171,37 @@ std::unique_ptr<PageReader> SerializedRowGroup::GetColumnPageReader(int i) {
       std::move(stream), FromThrift(col.meta_data.codec), properties_.allocator()));
 }
 
-RowGroupStatistics SerializedRowGroup::GetColumnStats(int i) const {
-  const format::ColumnMetaData& meta_data = metadata_->columns[i].meta_data;
+template <typename DType>
+static std::shared_ptr<RowGroupStatistics>
+MakeColumnStats(const format::ColumnMetaData& metadata, const ColumnDescriptor* descr) {
+  return std::make_shared<TypedRowGroupStatistics<DType>>(descr,
+      metadata.statistics.min, metadata.statistics.max, metadata.num_values,
+      metadata.statistics.null_count, metadata.statistics.distinct_count);
+}
 
-  RowGroupStatistics result;
-  result.num_values = meta_data.num_values;
-  result.null_count = meta_data.statistics.null_count;
-  result.distinct_count = meta_data.statistics.distinct_count;
-  result.max = &meta_data.statistics.max;
-  result.min = &meta_data.statistics.min;
-  return result;
+std::shared_ptr<RowGroupStatistics> SerializedRowGroup::GetColumnStats(int i) const {
+  const format::ColumnMetaData& meta_data = metadata_->columns[i].meta_data;
```

Comment:
This really belongs to a typed `ColumnReader`, but for now I just keep the old interface.